qa/tasks/cephfs/test_exports.py

```diff
@@ -517,3 +517,138 @@ def test_ephemeral_pin_shrink_mds(self):
         log.info("{0} migrations have occured due to the cluster resizing".format(count))
         # rebalancing from 3 -> 2 may cause half of rank 0/1 to move and all of rank 2
         self.assertLessEqual((count/len(subtrees_old)), (1.0/3.0/2.0 + 1.0/3.0/2.0 + 1.0/3.0)*1.25) # aka .66 with 25% overbudget
+
+class TestKillPoints(CephFSTestCase):
+    # Two active MDS
+    MDSS_REQUIRED = 4
+
+    init = False
+
+    def _setup_cluster(self):
+        # Set Multi-MDS cluster
+        self.fs.set_max_mds(2)
+
+        self.fs.wait_for_daemons()
+
+        # Create test data
+        if self.init is False:
+            self._populate_data(8)
+            self.init = True
+
+        return True
+
+    def _populate_data(self, nfiles):
+            size_mb = 8
+            dname = "abc"
+            self.mount_a.run_shell(["mkdir", dname])
+            for i in range(nfiles):
+                fname = dname + "/" + (str(i)) + ".txt"
+                self.mount_a.write_n_mb(fname, size_mb)
+
+            # Get the list of file to confirm the count of files
+            self.org_files = self.mount_a.ls()
+
+    def _verify_data(self):
+        s1 = set(self.mount_a.ls())
+        s2 = set(self.org_files)
+
+        if s1.isdisjoint(s2):
+            log.info("Directory contents org: %s" %str(self.org_files))
+            log.info("Directory contents new: %s" %str(self.mount_a.ls()))
+            return False
+
+        log.info("Directory contents matches")
+        return True
+
+    def _run_export_dir(self, importv, exportv):
+
+        status = self.fs.status()
+
+        rank_0 = self.fs.get_rank(status=status, rank = 0)
+        rank_1 = self.fs.get_rank(status=status, rank = 1)
+        rank_0_name = rank_0['name']
+        rank_1_name = rank_1['name']
+
+        killpoint = {}
+        killpoint[0] = ('export', exportv)
+        killpoint[1] = ('import', importv)
+
+        command = ["config", "set", "mds_kill_export_at", str(exportv)]
+        result = self.fs.rank_asok(command, rank=0, status=status)
+        assert(result["success"])
+
+        command = ["config", "set", "mds_kill_import_at", str(importv)]
+        result = self.fs.rank_asok(command, rank=1, status=status)
+        assert(result["success"])
+
+        try:
+            # This should kill either or both MDS process
+            self.mount_a.setfattr("abc", "ceph.dir.pin", "1")
```

Comment:
What's preventing the balancer from doing part of this export before reaching this point? (I think we need to pin the directory to rank 0 before populating it.)

Reply:
Good point. Yeah, actually we cannot be sure the `abc` dir is already pinned or auth in rank 0.

Reply:
> @lxbsz please run through QA to verify it works when you're done adjusting the code.
> 
Sure, will do that.